### PR TITLE
Upgrade to kotlin 1.4 & bug fixes for kotlin MPP

### PR DESCRIPTION
**Pull request by Thecrazyskull**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache [hunk 1]
```diff
@@ -6,28 +6,25 @@ package {{apiPackage}}
...
     serializer: KotlinxSerializer
 ) : ApiClient(baseUrl, httpClientEngine, serializer) {
 
-    @UseExperimental(UnstableDefault::class)
     constructor(
         baseUrl: kotlin.String = "{{{basePath}}}",
         httpClientEngine: HttpClientEngine? = null,
-        jsonConfiguration: JsonConfiguration = JsonConfiguration.Default
-    ) : this(baseUrl, httpClientEngine, KotlinxSerializer(Json(jsonConfiguration)))
+        json: Json = Json { }
```
Comment (dzolnai) [inline modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache:26]: I noticed that the app begins to complain if there are unknown keys in the response.
The fix would be to add ignoreUnknownKeywords in these braces. Can you add that?

**Review by dzolnai — CHANGES_REQUESTED** [review]
Almost good, one request

File: modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache (hunk 1 above)
Comment (Thecrazyskull) [reply modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache:26]: If that happens, it means that the response we get from the API does not align with the models. I'm not sure it's a good idea to silence such warnings. Do you still want me to do that?

Comment (dzolnai) [reply modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache:26]: An extra key or two is a backwards compatible change which could easily happen in a production setting. So let's turn it on.

Comment (Thecrazyskull) [reply modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/api.mustache:26]: fair enough. will do

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java [hunk 2]
```diff
@@ -780,6 +832,24 @@ protected boolean needToImport(String type) {
...
+        Map<String, Schema> allDefinitions = ModelUtils.getSchemas(this.openAPI);
+        if (allDefinitions != null) {
+            String parentSchema = m.parentSchema;
+
+            // multilevel inheritance: reconcile properties of all the parents
+            while (parentSchema != null) {
+                final Schema parentModel = allDefinitions.get(parentSchema);
+                final CodegenModel parentCodegenModel = super.fromModel(m.parent,
+                        parentModel);
+                System.out.println("parentCodegenModel="+parentCodegenModel);
```
Comment (dzolnai) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java:845]: Debug call?

**Review by dzolnai — APPROVED** [review]
Almost good

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java (hunk 2 above)
Comment (Thecrazyskull) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java:845]: oh, my bad

**Review by dzolnai — APPROVED** [review]

**Pull request merged**